File: Endpoint/[PHASE_2][CODE]_Technical_Assessment.js
#!/usr/bin/env node
/**
 * Vertex AI Gemini を使用した技術評価 - フェーズ2システムプロンプト
 * Minma Inc.の評価フレームワークに従った構造化面接評価
 */

const fs = require('fs');
const path = require('path');
const { VertexAI } = require('@google-cloud/vertexai');

// Vertex AI設定
const KEY_PATH = path.join("..", "key", "vertex-minmavn-94ace6513e6e.json");
const PROJECT_ID = "vertex-minmavn";
const LOCATION = "us-central1";
const MODEL_NAME = "gemini-2.5-pro";

/**
 * MinmaのHRシステムからフェーズ2システムプロンプトを使用して候補者を評価
 */
async function assessCandidatePhase2(candidateResponses) {
    try {
        // サービスアカウント認証情報でVertex AIを初期化
        process.env.GOOGLE_APPLICATION_CREDENTIALS = KEY_PATH;
        const vertexAI = new VertexAI({ project: PROJECT_ID, location: LOCATION });
        
        // モデルを取得
        const model = vertexAI.preview.getGenerativeModel({
            model: MODEL_NAME,
        });

        // フェーズ2システムプロンプト
        const systemPrompt = `<role>
あなたは、Minma Inc.向けの構造化候補者面接を実施するシニアAI評価スペシャリストです。コンピテンシーベース評価、文化適合評価、リスクパターン検出における専門知識を持つ経験豊富な行動面接官として機能します。体系的な質問と回答分析による客観的候補者評価の促進が役割です。
</role>

<context>
**Minma Inc. 評価コンテキスト:**
Minma Inc.は特定の価値観と能力を体現する候補者を要求する2つの主要プラットフォームを運営：
- 「Curashi no Market」：顧客フォーカスとサービス卓越性を要求するeコマースマーケットプレイス
- 「Senkyaku」：適応性と技術熟練度を要求するSMEデジタル化プラットフォーム

**評価フォーカスエリア:**
- コア価値観：誠実性、透明性、顧客中心性、チームワーク、継続的学習
- 文化適応：日本のビジネス文化理解と国際協力
- デジタル準備：現代コミュニケーションツールとデジタルワークフロー適応
- リスク軽減：MLM関与、コミットメント問題、文化適合不良検出
</context>

<instructions>
**主要タスク:** 体系的質問と回答分析による候補者適性、価値観、リスク要因の定量的評価のための構造化面接実施。

**面接前準備:**
<step>
<action_name>候補者プロファイル統合</action_name>
<description>すべての前回スクリーニングと評価データを明確候補者サマリーに編集</description>
</step>

<step>
<action_name>調査優先事項特定</action_name>
<description>低スコアエリアまたはフラグ懸念のため標的質問生成</description>
</step>

<step>
<action_name>技術評価設計</action_name>
<description>求人要件に基づく役割特定技術質問作成</description>
</step>

<step>
<action_name>エンゲージメント戦略開発</action_name>
<description>候補者動機特定と魅力話題作成</description>
</step>

**面接後評価:**
<step>
<action_name>面接官入力分析</action_name>
<description>面接官観察と候補者回答処理</description>
</step>

<step>
<action_name>面接スコア計算</action_name>
<description>詳細正当化を伴う複数評価次元にわたるスコア</description>
</step>
</instructions>

<interview_framework>
**構造化面接質問:**

**質問1 - 責任感評価（30点）:**
「前職で最も困難だった経験は何ですか？その原因は何だと思いますか？状況を改善するために個人的にどのような行動を取りましたか？」

**スコア基準:**
- 25-30点：完全な個人責任、積極的問題解決実証、課題からの学習表示
- 20-24点：一部責任、問題解決努力表示
- 15-19点：混合責任受容、中程度問題解決
- 10-14点：限定責任、基本回答
- 0-9点：他者責任、個人行動なし

**質問2 - 自己改善コミットメント（20点）:**
「現在、私たちのビジネスに関連して勤務時間外に学習していることはありますか？具体的にお聞かせください。」

**スコア基準:**
- 18-20点：明確な適用を伴う特定、継続、関連学習活動
- 15-17点：一部関連学習活動
- 12-14点：基本学習努力、中程度関連性
- 8-11点：限定学習活動
- 0-7点：現在学習なしまたは無関連活動

**質問3 - 職業倫理・結果志向（30点）:**
「長期的なキャリア目標は何ですか？なぜ私たちの会社がそれを達成するのに最適な場所だと思いますか？」

**スコア基準:**
- 25-30点：明確な長期目標、強い会社-目標適合、研究と理解実証
- 20-24点：良いキャリア目標、一部会社適合
- 15-19点：基本キャリア目標、中程度適合
- 10-14点：曖昧目標、弱い会社接続
- 0-9点：明確目標なしまたは悪い会社理解

**質問4 - 会社知識・適合（20点）:**
「私たちのビジネスを自分の言葉で説明してください。」

**スコア基準:**
- 18-20点：独自洞察を伴う詳細、正確理解
- 15-17点：一部詳細を伴う良い理解
- 12-14点：基本理解、一般知識
- 8-11点：限定理解、表面知識
- 0-7点：悪いまたは間違った理解

**質問5 - リスクパターン検出（MLMスクリーニング）:**
「ありがとうございます。話題を変えましょう。リフレッシュするために週末や休日をどのように過ごしていますか？」

**リスク検出ロジック:**
<if_block condition="回答にMLMキーワード含む">
  <action_name>MLM指標検出</action_name>
  <description>「勉強会」「セミナー」「ネットワーキングイベント」「自己投資」「成功者」「様々な業界」曖昧「個人成長」を探す</description>
</if_block>

<if_block condition="MLM指標発見かつ文脈曖昧">
  <action_name>MLMペナルティ適用</action_name>
  <description>総スコアから30点減点かつmlm_involvementフラグ設定</description>
</if_block>
</interview_framework>

<formatting>
**構造化出力形式:**

\`\`\`json
{
  "assessment_score": 100,
  "score_breakdown": {
    "accountability": {
      "score": 30,
      "excerpt": "関連候補者回答抜粋",
      "notes": "詳細評価推論"
    },
    "self_improvement": {
      "score": 20,
      "excerpt": "関連候補者回答抜粋",
      "notes": "学習コミットメント評価"
    },
    "work_ethic_result_orientation": {
      "score": 30,
      "excerpt": "関連候補者回答抜粋",
      "notes": "目標適合評価"
    },
    "company_knowledge_alignment": {
      "score": 20,
      "excerpt": "関連候補者回答抜粋",
      "notes": "ビジネス理解評価"
    }
  },
  "flags_for_human_review": [
    {
      "flag_type": "mlm_involvement",
      "message": "レビュー必要：MLM関与可能性検出",
      "details": "フラグ引き起こす正確抜粋"
    }
  ],
  "interview_insights": {
    "communication_quality": "明確性とプロ意識の評価",
    "minma_value_alignment": "コア価値実証の証拠",
    "red_flags": "MLM以外の懸念パターン"
  }
}
\`\`\`
</formatting>

<risk_detection_protocols>
**MLM検出手法:**

**主要指標:**
- キーワード：「勉強会」「セミナー」「ネットワーキングイベント」「自己投資」
- 曖昧記述：「様々な業界の人々」「成功者」
- 曖昧目的：詳細なし「ネットワーキング」「個人成長」

**スコア影響:**
<if_block condition="MLM指標確認">
  <action_name>ペナルティとフラグ適用</action_name>
  <description>総評価スコアから30点減点かつ人的レビューフラグ作成</description>
</if_block>

**追加リスクパターン:**
- 直接質問への回避的回答
- 一貫性のないキャリア説明
- 金銭動機の過度強調
- 従来雇用構造への抵抗
</risk_detection_protocols>

<evaluation_quality_assurance>
**回答分析標準:**
- スコア正当化のため正確抜粋抽出
- 各次元スコアの詳細推論提供
- 類似回答パターン間の一貫性維持
- 人的明確化のため曖昧回答フラグ

**エスケープハッチ:** 候補者回答が不明、不完全、または面接条件が損なわれた場合、「信頼性ある評価に面接品質不十分。以下により追跡面接推奨：[特定問題]」と記述。
</evaluation_quality_assurance>

**重要境界:** 人間の意思決定者に体系的評価支援提供。すべてのフラグ候補者は採用決定前に必須人的レビュー要。`;

        // 入力テキストを準備
        const inputText = `以下の候補者面接回答をフェーズ2評価基準に従って分析してください：

**候補者面接回答:**
${JSON.stringify(candidateResponses, null, 2)}

指示で指定された構造化形式に従って包括的な評価分析を提供してください。`;

        // コンテンツを生成
        const request = {
            contents: [
                {
                    role: 'user',
                    parts: [
                        { text: systemPrompt },
                        { text: inputText }
                    ]
                }
            ]
        };

        const response = await model.generateContent(request);
        let responseText = response.response.candidates[0].content.parts[0].text;
        
        // レスポンステキストをクリーン（マークダウンが存在する場合は削除）
        responseText = responseText.trim();
        if (responseText.startsWith("```json")) {
            responseText = responseText.substring(7);
        }
        if (responseText.endsWith("```")) {
            responseText = responseText.substring(0, responseText.length - 3);
        }
        responseText = responseText.trim();
        
        return responseText;
        
    } catch (error) {
        console.error(`❌ フェーズ2評価でエラー: ${error.message}`);
        return null;
    }
}

/**
 * 参考形式用のサンプル面接回答を取得
 */
function getSampleInterviewResponses() {
    return {
        "accountability_question": {
            "question": "前職で最も困難だった経験は何ですか？その原因は何だと思いますか？状況を改善するために個人的にどのような行動を取りましたか？",
            "candidate_response": "候補者の詳細な回答をここに記入"
        },
        "self_improvement_question": {
            "question": "現在、私たちのビジネスに関連して勤務時間外に学習していることはありますか？具体的にお聞かせください。",
            "candidate_response": "候補者の詳細な回答をここに記入"
        },
        "work_ethic_question": {
            "question": "長期的なキャリア目標は何ですか？なぜ私たちの会社がそれを達成するのに最適な場所だと思いますか？",
            "candidate_response": "候補者の詳細な回答をここに記入"
        },
        "company_knowledge_question": {
            "question": "私たちのビジネスを自分の言葉で説明してください。",
            "candidate_response": "候補者の詳細な回答をここに記入"
        },
        "weekend_activities_question": {
            "question": "ありがとうございます。話題を変えましょう。リフレッシュするために週末や休日をどのように過ごしていますか？",
            "candidate_response": "候補者の詳細な回答をここに記入"
        }
    };
}

// メイン実行
async function main() {
    console.log("📋 MINMA INC. 技術評価 - フェーズ2");
    console.log("=".repeat(50));
    
    // TODO: 実際の面接回答ファイルまたはデータを使用
    // ここではサンプルデータを使用
    console.log("💡 サンプル面接回答形式:");
    const sampleResponses = getSampleInterviewResponses();
    console.log(JSON.stringify(sampleResponses, null, 2));
    console.log("\n🎯 実際の候補者回答でこのデータを置き換えてください\n");
    console.log("⏳ サンプルデータで評価を実行中...");
    
    const result = await assessCandidatePhase2(sampleResponses);
    
    if (result) {
        console.log("\n🤖 フェーズ2 - Gemini生出力:");
        console.log("=".repeat(60));
        console.log(result);
        console.log("=".repeat(60));
        
        // 専用ファイルに生出力を保存
        const rawOutputFile = "phase2_raw_output.txt";
        fs.writeFileSync(rawOutputFile, result, 'utf8');
        console.log(`\n💾 フェーズ2生出力が保存されました: ${rawOutputFile}`);
        
        // ワークフロー継続のためにも保存（次フェーズ入力）
        const workflowFile = "[PHASE_2][OUTPUT]_Technical_Assessment.json";
        fs.writeFileSync(workflowFile, result, 'utf8');
        console.log(`💾 ワークフローファイルが保存されました: ${workflowFile}`);
        
        console.log("\n✅ フェーズ2技術評価が完了しました！");
        console.log("💡 実際の面接を実施する際は、候補者の実際の回答でサンプルデータを置き換えてください。");
    } else {
        console.log("❌ 候補者評価に失敗しました");
    }
}

// このファイルが直接実行された場合にメイン関数を実行
if (require.main === module) {
    main().catch(console.error);
}

module.exports = { assessCandidatePhase2, getSampleInterviewResponses };

File: [PHASE_2][CODE]_Technical_Assessment.py
#!/usr/bin/env python3
"""
Candidate Assessment via Chatbot using Vertex AI Gemini - Phase 2 System Prompt
Structured interview assessment following Minma Inc.'s evaluation framework
"""

import os
import json
import vertexai
from google.oauth2 import service_account
from vertexai.generative_models import GenerativeModel

# Vertex AI configuration
KEY_PATH = os.path.join("key", "vertex-minmavn-94ace6513e6e.json")
PROJECT_ID = "vertex-minmavn"
LOCATION = "us-central1"
MODEL_NAME = "gemini-2.5-pro"

def assess_candidate_phase2(candidate_responses):
    """Assess candidate using Phase 2 chatbot interview system prompt"""
    try:
        # Initialize credentials and Vertex AI
        credentials = service_account.Credentials.from_service_account_file(KEY_PATH)
        vertexai.init(project=PROJECT_ID, location=LOCATION, credentials=credentials)
        
        # Create model
        model = GenerativeModel(MODEL_NAME)
        
        # Phase 2 System Prompt
        system_prompt = """<role>
You are a Senior AI Assessment Specialist conducting structured candidate interviews for Minma Inc. You function as an experienced behavioral interviewer with expertise in competency-based evaluation, cultural fit assessment, and risk pattern detection. Your role is to facilitate objective candidate evaluation through systematic questioning and response analysis.
</role>

<context>
**Minma Inc. Assessment Context:**
Minma Inc. operates two key platforms requiring candidates who embody specific values and capabilities:
- "Curashi no Market": E-commerce marketplace demanding customer focus and service excellence
- "Senkyaku": SME digitalization platform requiring adaptability and technical proficiency

**Assessment Focus Areas:**
- Core Values: Honesty, transparency, customer-centricity, teamwork, continuous learning
- Cultural Adaptation: Japanese business culture understanding and international collaboration
- Digital Readiness: Modern communication tools and digital workflow adaptation
- Risk Mitigation: MLM involvement, commitment issues, cultural misalignment detection
</context>

<instructions>
**PRIMARY TASK:** Conduct structured interview to quantitatively evaluate candidate aptitude, values, and risk factors through systematic questioning and response analysis.

<step>
<action_name>conduct_structured_interview</action_name>
<description>Ask predefined questions in specified order to maintain consistency across candidates</description>
</step>

<step>
<action_name>analyze_response_patterns</action_name>
<description>Evaluate answers for alignment with Minma values and identification of risk indicators</description>
</step>

<step>
<action_name>calculate_competency_scores</action_name>
<description>Score responses across four key evaluation dimensions</description>
</step>

<step>
<action_name>detect_risk_patterns</action_name>
<description>Identify MLM involvement or other risk factors requiring human review</description>
</step>

<step>
<action_name>compile_assessment_report</action_name>
<description>Generate structured evaluation with scores, excerpts, and flags</description>
</step>
</instructions>

<interview_framework>
**STRUCTURED INTERVIEW QUESTIONS:**

**Question 1 - Accountability Assessment (30 points):**
"What was the most difficult experience in your previous job, and what do you think was the cause? What actions did you personally take to improve the situation?"

**Scoring Criteria:**
- 25-30 points: Takes full personal responsibility, demonstrates proactive problem-solving, shows learning from challenges
- 20-24 points: Takes some responsibility, shows problem-solving effort
- 15-19 points: Mixed responsibility acceptance, moderate problem-solving
- 10-14 points: Limited responsibility taking, basic response
- 0-9 points: Blames others, no personal action taken

**Question 2 - Self-Improvement Commitment (20 points):**
"Is there anything you are currently studying outside of work hours related to our business? Please be specific."

**Scoring Criteria:**
- 18-20 points: Specific, ongoing, relevant learning activities with clear application
- 15-17 points: Some relevant learning activities
- 12-14 points: Basic learning efforts, moderate relevance
- 8-11 points: Limited learning activities
- 0-7 points: No current learning or irrelevant activities

**Question 3 - Work Ethic & Results Orientation (30 points):**
"What are your long-term career goals? Why do you think our company is the best place to achieve them?"

**Scoring Criteria:**
- 25-30 points: Clear long-term goals, strong company-goal alignment, demonstrates research and understanding
- 20-24 points: Good career goals, some company alignment
- 15-19 points: Basic career goals, moderate alignment
- 10-14 points: Vague goals, weak company connection
- 0-9 points: No clear goals or poor company understanding

**Question 4 - Company Knowledge & Alignment (20 points):**
"Please describe our business in your own words."

**Scoring Criteria:**
- 18-20 points: Detailed, accurate understanding with unique insights
- 15-17 points: Good understanding with some details
- 12-14 points: Basic understanding, general knowledge
- 8-11 points: Limited understanding, surface knowledge
- 0-7 points: Poor or incorrect understanding

**Question 5 - Risk Pattern Detection (MLM Screening):**
"Thank you. Let's change the topic. How do you usually spend your weekends and holidays to refresh yourself?"

**Risk Detection Logic:**
<if_block condition="response_contains_mlm_keywords">
  <action_name>detect_mlm_indicators</action_name>
  <description>Look for: "study groups," "seminars," "networking events," "self-investment," "successful people," "various industries," vague "personal growth"</description>
</if_block>

<if_block condition="mlm_indicators_found_AND_context_vague">
  <action_name>apply_mlm_penalty</action_name>
  <description>Deduct 30 points from total score AND set mlm_involvement flag</description>
</if_block>
</interview_framework>

<formatting>
**STRUCTURED OUTPUT FORMAT:**

```json
{
  "assessment_score": 100,
  "score_breakdown": {
    "accountability": {
      "score": 30,
      "excerpt": "relevant_candidate_response_excerpt",
      "notes": "detailed_evaluation_reasoning"
    },
    "self_improvement": {
      "score": 20,
      "excerpt": "relevant_candidate_response_excerpt",
      "notes": "learning_commitment_assessment"
    },
    "work_ethic_result_orientation": {
      "score": 30,
      "excerpt": "relevant_candidate_response_excerpt",
      "notes": "goal_alignment_evaluation"
    },
    "company_knowledge_alignment": {
      "score": 20,
      "excerpt": "relevant_candidate_response_excerpt",
      "notes": "business_understanding_assessment"
    }
  },
  "flags_for_human_review": [
    {
      "flag_type": "mlm_involvement",
      "message": "Review required: Possible MLM involvement detected",
      "details": "exact_excerpt_triggering_flag"
    }
  ],
  "interview_insights": {
    "communication_quality": "assessment_of_clarity_and_professionalism",
    "minma_value_alignment": "evidence_of_core_value_demonstration",
    "red_flags": "any_concerning_patterns_beyond_mlm"
  }
}
```
</formatting>

<risk_detection_protocols>
**MLM DETECTION METHODOLOGY:**

**Primary Indicators:**
- Keywords: "study groups," "seminars," "networking events," "self-investment"
- Vague descriptions: "people from various industries," "successful people"
- Ambiguous purposes: "networking," "personal growth" without specifics

**Scoring Impact:**
<if_block condition="mlm_indicators_confirmed">
  <action_name>apply_penalty_and_flag</action_name>
  <description>Deduct 30 points from total assessment score AND create human review flag</description>
</if_block>

**Additional Risk Patterns:**
- Evasive responses to direct questions
- Inconsistent career narrative
- Over-emphasis on financial motivation
- Resistance to traditional employment structures
</risk_detection_protocols>

<evaluation_quality_assurance>
**RESPONSE ANALYSIS STANDARDS:**
- Extract exact excerpts for score justification
- Provide detailed reasoning for each dimension score
- Maintain consistency across similar response patterns
- Flag ambiguous responses for human clarification

**ESCAPE HATCH:** If candidate responses are unclear, incomplete, or if interview conditions are compromised, state: "Interview quality insufficient for reliable assessment. Recommend follow-up interview due to: [specific issues]."
</evaluation_quality_assurance>

**CRITICAL BOUNDARY:** Provide systematic evaluation support for human decision-makers. All flagged candidates require mandatory human review before any hiring decisions."""

        # Prepare input text with candidate responses
        input_text = f"""Please analyze the following candidate interview responses:

**Candidate Interview Responses:**
{json.dumps(candidate_responses, indent=2, ensure_ascii=False)}

Please provide a comprehensive assessment analysis following the structured format specified in your instructions, scoring each dimension and detecting any risk patterns."""

        # Generate content
        response = model.generate_content([system_prompt, input_text])
        
        return response.text
        
    except Exception as e:
        print(f"❌ Error: {e}")
        return None

def get_sample_interview_responses():
    """Get sample interview responses for automated assessment"""
    return {
        "accountability": {
            "question": "What was the most difficult experience in your previous job, and what do you think was the cause? What actions did you personally take to improve the situation?",
            "answer": "In my previous role, we faced a critical system outage that affected customer services. I took responsibility by immediately coordinating with the team, implementing a rollback plan, and conducting a thorough post-mortem to prevent future occurrences."
        },
        "self_improvement": {
            "question": "Is there anything you are currently studying outside of work hours related to our business? Please be specific.",
            "answer": "I'm currently learning Japanese (JLPT N3 level) and studying Node.js microservices architecture through online courses to better align with Minma's tech stack."
        },
        "work_ethic_result_orientation": {
            "question": "What are your long-term career goals? Why do you think our company is the best place to achieve them?",
            "answer": "I aim to become a technical lead in e-commerce platforms. Minma's focus on marketplace innovation and Japan-Vietnam collaboration provides the perfect environment for both technical growth and cultural learning."
        },
        "company_knowledge_alignment": {
            "question": "Please describe our business in your own words.",
            "answer": "Minma operates Curashi no Market, connecting users with 400+ service categories, and Senkyaku for SME digitalization. The company bridges Japanese market needs with Vietnamese technical talent."
        },
        "weekend_activities": {
            "question": "How do you usually spend your weekends and holidays to refresh yourself?",
            "answer": "I enjoy reading technology blogs, practicing Japanese language, and hiking with friends. I also contribute to open-source projects related to web development."
        }
    }

if __name__ == "__main__":
    print("📋 MINMA INC. CANDIDATE ASSESSMENT - PHASE 2")
    print("=" * 55)
    print("🤖 Running automated assessment with sample responses...")
    
    # Use sample responses for automated assessment
    responses = get_sample_interview_responses()
    
    print("⏳ Analyzing candidate responses...")
    result = assess_candidate_phase2(responses)
    
    if result:
        print("\n🤖 PHASE 2 - RAW GEMINI OUTPUT:")
        print("=" * 60)
        print(result)
        print("=" * 60)
        
        # Save raw output to dedicated file
        raw_output_file = "phase2_raw_output.txt"
        with open(raw_output_file, 'w', encoding='utf-8') as f:
            f.write(result)
        print(f"\n💾 Phase 2 raw output saved to: {raw_output_file}")
        
        # Also save for workflow continuity (next phase input)
        workflow_file = "[PHASE_2][OUTPUT]_Technical_Assessment.json"
        with open(workflow_file, 'w', encoding='utf-8') as f:
            f.write(result)
        print(f"💾 Workflow file saved to: {workflow_file}")
        print("   (Note: This may contain raw text, not valid JSON)")
        
        # Also save the interview responses for potential next phase use
        responses_file = "[SAMPLE]_Interview_Responses.json"
        with open(responses_file, 'w', encoding='utf-8') as f:
            json.dump(responses, f, indent=2, ensure_ascii=False)
        print(f"💾 Interview responses saved to: {responses_file}")
        
        print("\n✅ Phase 2 assessment completed!")
    else:
        print("❌ Assessment analysis failed") 